cli_helpers/tabular_output/preprocessors.py

```diff
@@ -83,6 +83,7 @@ def align_decimals(data, headers, column_types=(), **_):
 
     """
     pointpos = len(headers) * [0]
+    data = list(data)
```

Comment:
I almost wonder if `format_output()` should pass a `list` to the preprocessors. If every preprocessor is going to have to create a list out of the iterator, wouldn't it be better to just do it in one spot?